### PR TITLE
IntentTEDPolicy: Sort rasa test output by severity

### PR DESCRIPTION
**Pull request by dakshvar22**
**Proposed changes**:
- Closes https://github.com/RasaHQ/rasa/issues/8970

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by alwx** [comment]
Updated code style, checked the implementation of the `_sort_trackers_with_severity_of_warning` (and made some small updates to it), and wrote a test for sorting trackers.
@wochinge please, check

File: changelog/8970.improvement.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Adapt `rasa test` to sort `IntentTEDPolicy` warnings by severity.
```
Comment (wochinge) [inline changelog/8970.improvement.md:1]: is this behavior documented in the docs @dakshvar22 ?

File: rasa/core/test.py [hunk 2]
```diff
@@ -324,6 +325,15 @@ def inline_comment(self) -> Text:
     def as_story_string(self) -> Text:
         return f"{self.action_name}   <!-- {self.inline_comment()} -->"
 
+    def __repr__(self) -> Text:
```
Comment (wochinge) [inline rasa/core/test.py:328]: 🎉

File: rasa/core/test.py [hunk 3]
```diff
@@ -800,6 +811,45 @@ def _in_training_data_fraction(action_list: List[Dict[Text, Any]]) -> float:
...
+                    event.metadata.get(QUERY_INTENT_KEY, {}).get(SEVERITY_KEY, 0),
+                )
+        tracker_severity_scores.append(max_severity)
+
+    sorted_trackers_with_severity = sorted(
+        [
+            (severity, tracker)
+            for severity, tracker in zip(tracker_severity_scores, trackers_to_sort)
+        ],
+        key=lambda x: -x[0],
```
Comment (wochinge) [inline rasa/core/test.py:847]: can please use a more descriptive parameter name than `x`?

File: rasa/core/test.py [hunk 4]
```diff
@@ -800,6 +811,45 @@ def _in_training_data_fraction(action_list: List[Dict[Text, Any]]) -> float:
...
+                and event.action_name_prediction == ACTION_UNLIKELY_INTENT_NAME
+            ):
+                max_severity = max(
+                    max_severity,
+                    event.metadata.get(QUERY_INTENT_KEY, {}).get(SEVERITY_KEY, 0),
+                )
+        tracker_severity_scores.append(max_severity)
+
+    sorted_trackers_with_severity = sorted(
+        [
```
Comment (wochinge) [inline rasa/core/test.py:843]: why do we need the list comprehension around it?

File: tests/core/test_test.py [hunk 5]
```diff
@@ -1,18 +1,30 @@
...
+    SCORE_KEY,
+)
+from rasa.core.constants import STORIES_WITH_WARNINGS_FILE
 
 
-def _probabilities_with_action_unlikely_intent_for(intent_name: Text):
+def _probabilities_with_action_unlikely_intent_for(
+    intent_names: List[Text],
+    metadata_for_intent: Optional[Dict[Text, Dict[Text, Any]]] = None,
+):
```
Comment (wochinge) [inline tests/core/test_test.py:27]: can we please add the return type here?

**Review by wochinge — COMMENTED** [review]
Cool idea 💯

File: changelog/8970.improvement.md (hunk 1 above)
Comment (dakshvar22) [reply changelog/8970.improvement.md:1]: Yes, it is.

**Review by wochinge — APPROVED** [review]
🚀

**Pull request merged**